### PR TITLE
Added mod module proposal

### PR DESCRIPTION
**Pull request by 2767mr**

### CODE REVIEW

**Review by streetclaw — APPROVED** [review]
Sounds good.

**Comment by 2767mr** [comment]
We still need to discuss which solution to actually pick

File: proposals/1.0.0/modModules.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+It is possible for a modloader to hook require add a node_module search path for each mod by inspecting the `document.currentScript` property. This can allow for mods to have their own separate modules.
+
+#### Advantages
+
+* Does not cause interference with other mods
+
+#### Disadvantages
+
+* Can cause a lot of bloat because each mod needs to come with all dependencies
+* Does not work with delayed requires because `document.currentScript` will loose it's value.
```
Comment (ghost) [inline proposals/1.0.0/modModules.md]: lose

File: proposals/1.0.0/modModules.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* It will always find a installed module
+* If implemented as a global folder it can reduce bloat
+
+#### Disadvantages
+
+* Mods can cause interference if the same module with a different version is installed
+
+### Per mod with fallback
+
+A modloader can used the method described in "Module resolution per mod" until `document.currentScript` looses it's value and then fall back to the method of "Global modules".
```
Comment (ghost) [inline proposals/1.0.0/modModules.md]: loses

**Review by ghost — COMMENTED** [review]
Format looks good.

**Review by ghost — APPROVED** [review]

**Pull request merged**